samples/csharp_dotnetcore/11.qnamaker/QnAMaker/BotServices.cs

```diff
@@ -19,10 +20,51 @@ public class BotServices
         /// <summary>
         /// Initializes a new instance of the <see cref="BotServices"/> class.
         /// </summary>
-        /// <param name="qnaServices">A dictionary of named <see cref="QnAMaker"/> instances for usage within the bot.</param>
-        public BotServices(Dictionary<string, QnAMaker> qnaServices)
+        /// <param name="botConfiguration">A dictionary of named <see cref="BotConfiguration"/> instances for usage within the bot.</param>
+        public BotServices(BotConfiguration botConfiguration)
         {
-            QnAServices = qnaServices ?? throw new ArgumentNullException(nameof(qnaServices));
+            foreach (var service in botConfiguration.Services)
```

Comment:
As an aside, we just add a bunch of these validations into the 4.3 (soon!) release.

Reply:
Looking forward to it!